### PR TITLE
create grant flow filter

### PR DESCRIPTION
**Pull request by aryszka**
**[WIP]**

Signed-off-by: Arpad Ryszka <arpad.ryszka@gmail.com>

### CODE REVIEW

File: filters/auth/grantflowstate.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		return "", err
+	}
+
+	nonce, err := encrypter.CreateNonce()
+	if err != nil {
+		return "", err
+	}
+
+	state := state{
+		Rand:       randString(randomStringLength),
```
Comment (muaazsaleem) [inline filters/auth/grantflowstate.go:51]: We should use `crypto` based `rand` here

File: filters/auth/grantflowstate.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	"fmt"
+	"time"
+
+	"github.com/zalando/skipper/secrets"
+)
+
+const randomStringLength = 20
+
+type state struct {
+	Rand       string `json:"rand"`
```
Comment (szuecs) [inline filters/auth/grantflowstate.go:15]: As discussed we can remove this field

**Comment by soceanainn** [comment]
As a backend engineer - the possibility of returning some raw HTML from an API endpoint and having Skipper handling all of the OAuth2 login flow is something that's very exciting as it opens up a lot of possibilities for our team internally 👍

**Comment by aryszka** [comment]
closing this as it was finalized in: https://github.com/zalando/skipper/pull/1609
Thanks for the contributors!

**Pull request closed without merging**